Business/Concrete/ProductManager.cs

```diff
@@ -7,35 +12,55 @@ namespace WebNetSample.Business.Concrete;
 
 public class ProductManager : IProductService
 {
-    private IProductRepository _productRepository;
+    private readonly IProductRepository _productRepository;
+    private readonly ICategoryRepository _categoryRepository;
+    private readonly ISupplierRepository _supplierRepository;
 
-    public ProductManager(IProductRepository productRepository)
+    public ProductManager(IProductRepository productRepository, ICategoryRepository categoryRepository,
+        ISupplierRepository supplierRepository)
     {
         _productRepository = productRepository;
+        _categoryRepository = categoryRepository;
+        _supplierRepository = supplierRepository;
     }
 
-    public Task Add(Product product) => _productRepository.AddAsync(product);
+    public Task AddAsync(Product product) =>
+        _productRepository.AddAsync(product);
 
-    public void Delete(Product product) => _productRepository.Delete(p => p.Id == product.Id);
+    public void Delete(Product product) =>
+        _productRepository.Delete(p => p.Id == product.Id);
 
-    public void Update(Product product) => _productRepository.Update(product);
+    public void Update(Product product) =>
+        _productRepository.Update(product);
 
     public async Task<Product> GetByIdAsync(Guid productId) => 
         await _productRepository.GetAsync(entity => entity.Id == productId);
 
-    public async Task<List<Product>> GetListAsync() => 
-        await _productRepository.GetListAsync();
+    public async Task<List<Product>> GetListAsync(PaginationParameters paginationParameters) =>
+        _productRepository.GetListAsync().Result
+            .Skip((paginationParameters.RecordsToSkip))
+                .Take(paginationParameters.PageSize).ToList();
 
     public async Task<List<Product>> GetListByCategoryIdAsync(Guid categoryId) => 
         await _productRepository.GetListAsync(entity => entity.CategoryId == categoryId);
 
-    public async Task<List<ProductDetailDto>> GetProductDetailsAsync() => 
-        await _productRepository.GetProductDetailsAsync();
-
-    public async Task<List<ProductDetailDto>> GetProductDetailsByCategoryIdAsync(Guid categoryId) => 
-        await _productRepository.GetProductDetailsAsync(entity => entity.CategoryId == categoryId);
+    public async Task<List<ProductDetailDto>> GetProductDetailsAsync(Expression<Func<Product, bool>> filter = null)
+    {
+        var result = (from product in _productRepository.GetListAsync().Result
```

Comment:
Sorry, but I do not really understand why do move collections joining to business layer. You previous solution is correct, the only one issue was related incorrect type usage - it should be entity instead of DTO. Your current solution not too good because it will be a serious performance issue. You are loading to server **all** entities from 3 different tables and after that you are only joining them. Your server will have to store a lot of items that will not be used in there is no joining pair. Please, revert your previous solution with joining entities on DB side and aggregate this join into `entity` model. It will be ok

Reply:
But this is a joined table, how can i return a just product entity?

Reply:
You can simplify add new model into `Entities` that will store aggregated result of 3 different tables

Reply:
Can you check [this](https://github.com/osmanaliyardim/WebNetSample/pull/4/commits/373de8feafd821a34d97c9209d5d18866f537e77) commit?

Reply:
Also check [this](https://github.com/osmanaliyardim/WebNetSample/pull/4/commits/288205d41484de3893403e35f95b1af258137af5) please